### ISSUE TITLE
build: Use tarball instead of dmg for macOS

### ISSUE THREAD

**Issue by dongcarl** [bitcoin/bitcoin#18128]
Since macOS is able to to extract tarballs with a double-click, I'm wondering what people's thoughts are on using tarballs instead of dmg for macOS builds.

This will eliminate the following build dependencies:
- native_biplist
- native_cdrkit
- native_ds_store
- native_libdmg-hfsplus
- native_mac_alias
- librsvg2-bin
- libtiff-tools
- imagemagick
- fonts-tuffy

I would say that this is a substantial enough list to seriously consider the tarball option.

A valid concern is one of UX, which might be worse.

Here were the steps to installing Bitcoin in a DMG world:

1. Double-click the DMG
2. Drag the .app to the aliased Applications folder (with an arrow as the hint)

tarball world steps:

1. Double-click the tarball
2. Drag the .app to Applications folder (no hint)

ping @fanquake

**Comment by jonasschnelli** [comment]
My 5cts on this:
* Using the .zip file seems to be an approach done more and more by common software packages
* The disk image offers a more native way to "install" an app by showing the Application folder (+ the arrow)
* Additional, the disk image allows for branding space.

I personally would ack a move from .dmg to zip. I don't think we have to deal with users not knowing how to move an .app bundle form the downloads folder to the application folder.

Additionally, it (the .zip approach) may simplify macOS app notarization.

**Comment by dongcarl** [comment]
Ah yes, a `.zip` might be better... From [the link you posted on IRC](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution#3087722):

> You can notarize an existing disk image, installer package, or ZIP archive containing your app.

**Comment by ch4ot1c** [comment]
I'd suggest to stick with the .dmg, just because some users really don't know that .apps should go in Applications/.

Though, your point makes sense. Zipping the .dmg isn't a particularly good option either. Maybe include a README with 'installation steps' (one move) in the zipped .app?

**Comment by fjahr** [comment]
I agree with @ch4ot1c that some users may not know that the app needs to be moved to `Applications/` but I think that could be solved with a clear instruction to do so next to the download link. If people don't even read one-liner instructions then we can't help them. I think that also has a better chance of being read than a README file in the zip, the concept of a plain text README file is pretty weird to non-programmers. And I also feel like I have seen this more lately, as @jonasschnelli says, so I think it will not be foreign to most users. So Concept ACK from me.

**Comment by fanquake** [comment]
> Using the .zip file seems to be an approach done more and more by common software packages

@jonasschnelli Can you give some examples? I had a quick look around, and Firefox, Google Chrome, Tor, VLC, Transmission, VirtualBox etc all still use a `.dmg`.

I'm still not entirely sure, but could probably be convinced that the reduction in build-system complexity is worth the trade-off in "branding". I might put those changes together as an alternative to/alongside #18151.

**Comment by fanquake** [comment]
After some more thought I think this is the way forward, and have pretty much finished up the changes required to switch to using a `.zip`. I'll PR them shortly and close #18151.

**Comment by laanwj** [comment]
~0 on this. I like the simplification in build system complexity, but on the other hand,  `.dmg` is what most people expect for macOS software and as far as I know, by far most software uses that as distribution.

**Comment by dongcarl** [comment]
GitHub Desktop uses zips instead of dmgs: https://desktop.github.com/

**Comment by sheffine** [comment]
> GitHub Desktop uses zips instead of dmgs: https://desktop.github.com/

Thank you

**Comment by sheffine** [comment]
Hello ,
Can I get the guide for this I am pretty new and have no idea how this work
,

Thanks

**Comment by jarolrod** [comment]
my 2cnts, the move to a `.zip` is enticing in that we can eliminate some build dependencies. But, here are some arguments against that move:
- `.dmg` is still the dominant format for distributing application on the macOS platform. It offers instant assurance that the application I just downloaded is a `macOS` specific application. Receiving a `.zip` may cast some level of doubt, as in: Did the server mess up and feed me a Windows application in a `.zip`
- It sucks that Apple always chooses to do it's own thing instead of adopting industry standards, but `.dmg` is still the standard and we should continue to use it for the foreseeable future since it is still what people expect.

**Comment by prusnak** [comment]
We simplified the creation of `dmg` in https://github.com/bitcoin/bitcoin/pull/23909 removing the half of the dependencies from the OP:

This will eliminate the following build dependencies:

- native_biplist
- native_cdrkit
- native_ds_store
- native_libdmg-hfsplus
- native_mac_alias
- ~librsvg2-bin~
- ~libtiff-tools~
- ~imagemagick~
- ~fonts-tuffy~

Do we still want to consider moving from `dmg` to `zip`?

**Comment by fanquake** [comment]
> Do we still want to consider moving from dmg to zip?

I don't think we'll make the move from dmg to zip yet, but I have opened another PR (#24031) which I think makes an OK tradeoff in regards to removing additional dependencies / build complexity from the macOS build.

**Comment by fanquake** [comment]
Updated the op to refect build changes since it was first written.
